test/e2e/conftest.py

```
@@ -471,13 +474,21 @@ def operating_system() -> OperatingSystem:
 
 def pytest_collection_modifyitems(items):
     sorted_list = list(items)
+    session_worker_tests = []
+    asset_sync_session_worker_tests = []
+
     for item in items:
-        # Run session scoped tests last to prevent Worker conflicts with class and function scoped tests.
-        if (
-            "session_worker" in item.fixturenames
-            or "asset_sync_session_worker" in item.fixturenames
-        ):
+        # Separate session worker tests into two groups to prevent worker conflicts
+        if "asset_sync_session_worker" in item.fixturenames:
             sorted_list.remove(item)
-            sorted_list.append(item)
+            asset_sync_session_worker_tests.append(item)
+        elif "session_worker" in item.fixturenames:
+            sorted_list.remove(item)
+            session_worker_tests.append(item)
```

can we raise an error here if for some reason a test requests both fixtures?

---

Good catch! I don't think this fixture will live for too long, but it's always good to code defensive and fail fast. Addressed in the rev.